thorlcr/master/thmastermain.cpp

```diff
@@ -1162,18 +1162,23 @@ int main( int argc, const char *argv[]  )
     }
     if (isContainerized())
     {
+        int retCode = exception ? TEC_Exception : 0;
         if (!cloudJobName.isEmpty())
         {
+            if (exception)
+            {
+                Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+                Owned<IConstWorkUnit> wu = factory->openWorkUnit(workunit);
+                if (wu)
+                {
+                    relayWuidException(wu, exception);
+                    retCode = 0; // if successfully reported, suppress thormanager exit failure that would trigger another exception
```

Comment:
If the exit failure is suppressed, does it mean that some error conditions will no longer be resolved thor manager exit failure handling (e.g. restarting thor)?   i.e., does it mean continuing as if no error occurred, mean that the error condition could re-occur when the next job is processed?

Reply:
This is containerized only, thor doesn't restart in k8s, existing instances are either reused or a new instance is spun up.
So this already reported error (to workunit) is suppressed to avoid the launcher of the instance (agentexec), which may be from a much earlier workunit, seeing the error, and reporting a k8s error "Job has reached the specified backoff limit".

In other words, if it gets this far, this thor instance has handled the error, reported it to our workflow, and is returning success to the agent to prevent it to report a spurious and confusing additional error.

Reply:
@shamser - please see reply.